variation-commons-core/src/main/java/uk/ac/ebi/eva/commons/core/models/factories/VariantVcfFactory.java

```diff
@@ -351,4 +361,38 @@ protected static int mapToMultiallelicIndex(int parsedAllele, int numAllele) {
         }
         return correctedAllele;
     }
+
+    private void checkVariantInformation(Variant variant, String fileId,
+                                           String studyId) throws NonVariantException, IncompleteInformationException {
+
+        // check genotypes
+        VariantSourceEntry variantSourceEntry = variant.getSourceEntry(fileId, studyId);
+        List<Map<String, String>> samplesData = variantSourceEntry.getSamplesData();
+        if (!samplesData.isEmpty()) {
+            if (samplesData.stream().map(m -> m.get("GT")).noneMatch(this::genotypeHasAlternateAllele)) {
+                throw new NonVariantException("TODO MESSAGE HERE");
+            }
+        } else {
+            // check info
+            boolean hasAC = checkAttributeIsNotZero(variantSourceEntry, "AC");
+            boolean hasAF = checkAttributeIsNotZero(variantSourceEntry, "AF");
+            boolean hasAN =  checkAttributeIsNotZero(variantSourceEntry, "AN");
+            boolean hasGTC =  checkAttributeIsNotZero(variantSourceEntry, "GTC");
+            if (!hasAC && !hasAF && !hasAN && !hasGTC) {
```

Comment:
does it make sense to not fail if we get AN only? or AC only?

Reply:
that is true, the allele frequency can be calculated just if we have both AN and AC. I will make it fail if just some of them are present.